012-OFEP-variations-and-variables.md

```diff
@@ -0,0 +1,167 @@
+```
+
+### Variations
+
+We now enter the realm of A/B testing and experimentation.
+
+An experiment consists of different variations (which are usually of `String` type), and the user is consistently bucketed (assigned) into one of its variations, irrespective of the device or session the user is in.
+
+You can think of an experiment named `my_experiment`, that can have two variations:
+
+- `control`: the default behaviour
+- `treatment`: the new behaviour that we wish to test
+
+A user will be bucketed into one of these variations, and will stay in that variation for the duration of the experiment.
+
+Example:
+
+```js
+const featureKey = "my_experiment";
+const defaultVariation = "control";
+
+const variation = client.getVariation(featureKey, defaultVariation);
+```
+
+#### Weights
+
+Each variation is also associated with a weight, which is a number between 0 (0%) and 1 (100%), and represents the probability of the user being bucketed into that variation. The total weight of all variations in an experiment must be 1.
+
+This level of details is something to be handled at Provider level, and not the OpenFeature SDKs.
+
+#### Measuring success
+
+Applications will be tracking various different conversion goals (like successful user sign ups, purchases, etc). Based on the variation the user is bucketed into, we will be able to track the conversion goal events against each variation of the experiment over a period of time, and decide which variation is the winner.
+
+OpenFeature SDKs will not be involved here.
+
+### Variables
+
+Feature management can go beyond just on/off values for the flag itself and supporting a mechanism for a/b testing with variations.
+
+We may also want to support the concept of variables for supporting configuration to be deployed independently of our applications, which are values that can be changed over time.
+
+These values can be of various different primitive types for different use cases.
+
+You can think of a feature named `my_feature`, that can have several variables scoped under it:
+
+- `showSidebar`: boolean
+- `buttonColor`: string
+- `allowSignup`: boolean
+
+The variable values can be evaluated against the context that already exists in the SDK, and can also react further based on the bucketed variation if there's an experiment involved.
+
+Example:
+
+```js
+const featureKey = "my_feature";
+const variableKey = "showSidebar";
+const defaultValue = false;
+
+const showSidebar = client.getVariableBoolean(featureKey, variableKey, defaultValue);
+```
+
+## Summarizing the feature model
+
+Three different kinds of values under a single feature:
+
+- Flag status (`Boolean`)
+- Variation (`String`)
+- Variables:
+  - key (`String`)
+  - Value (`Boolean`, `String`, `Number`, `Object`)
+
+**Note**: It is worthwhile to expand the `Number` type into more explicit types like `Integer` and `Double`, especially when dealing with SDKs in strongly typed languages.
```

Comment:
This is already supported in strongly typed languages.

Reply:
I will remove this line soon.